### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by virtushawk**
module 4

### CODE REVIEW

File: controller/src/main/java/com/epam/esm/exception/CustomExceptionHandler.java [hunk 1]
```diff
@@ -43,12 +43,13 @@
...
+    /*@ExceptionHandler(RuntimeException.class)
     public ResponseEntity<Object> handleInternalServerError(RuntimeException exception, Locale locale) {
-        String errorMessage = messageSource.getMessage(INTERNAL_SERVER_CODE, new Object[]{}, locale);
-        return new ResponseEntity<>(createErrorResponse(errorMessage, INTERNAL_SERVER_ERROR_CODE),
+        Map<Object, String> message = new HashMap<>();
+        message.put(ERROR_MESSAGE, messageSource.getMessage(INTERNAL_SERVER_CODE, new Object[]{}, locale));
+        return new ResponseEntity<>(createCustomErrorResponse(message, INTERNAL_SERVER_ERROR_CODE),
                 HttpStatus.INTERNAL_SERVER_ERROR);
-    }
+    }*/
```
Comment (maximelyashevich) [inline controller/src/main/java/com/epam/esm/exception/CustomExceptionHandler.java]: Please review this exception handler.

File: controller/src/main/java/com/epam/esm/util/ResponseAssembler.java [hunk 2]
```diff
@@ -138,6 +138,10 @@ private static TagDTO mapTag(TagDTO tagDTO) {
     }
 
     private static UserDTO mapUser(UserDTO userDTO) {
+        if (!ObjectUtils.isEmpty(userDTO.getOrders())) {
```
Comment (maximelyashevich) [inline controller/src/main/java/com/epam/esm/util/ResponseAssembler.java:141]: What's about `org.apache.commons.lang3.ObjectUtils#isNotEmpty` method?

File: spring-boot-admin-server/src/main/java/com/epam/esm/config/WebSecurityConfig.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+//import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+//import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+//import org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler;
+//import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+//import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+//
+//import java.util.UUID;
+//
+//@Configuration
+//public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
```
Comment (maximelyashevich) [inline spring-boot-admin-server/src/main/java/com/epam/esm/config/WebSecurityConfig.java]: Please review this class.

**Review by maximelyashevich — APPROVED** [review]
Good job!

**Review by ykazlovich — APPROVED** [review]

File: controller/src/main/java/com/epam/esm/exception/CustomExceptionHandler.java (hunk 1 above)
Comment (virtushawk) [reply controller/src/main/java/com/epam/esm/exception/CustomExceptionHandler.java]: done!

File: controller/src/main/java/com/epam/esm/util/ResponseAssembler.java (hunk 2 above)
Comment (virtushawk) [reply controller/src/main/java/com/epam/esm/util/ResponseAssembler.java:141]: I believe they have the same functionality, but i changed to the apache class

File: spring-boot-admin-server/src/main/java/com/epam/esm/config/WebSecurityConfig.java (hunk 3 above)
Comment (virtushawk) [reply spring-boot-admin-server/src/main/java/com/epam/esm/config/WebSecurityConfig.java]: Done! I deleted this class for now

**Pull request merged**